### ISSUE TITLE
Assert by compute generates ill-typed AIR

### ISSUE THREAD

**Issue by matthias-brun** [verus-lang/verus#294]
```rust
use builtin::*;
use builtin_macros::*;
mod pervasive;
use pervasive::*;
use seq::*;

verus! {

struct S {
    s: Seq<nat>,
}

impl S {
    spec fn f(self, idx: nat) -> nat {
        self.s.index(idx)
    }
}

proof fn g() {
    let x = S { s: seq![1,2,3] };
    assert(x.f(0) == 1) by(compute);
}

fn main() {
}
}
```

```
thread 'rustc' panicked at 'internal error: generated ill-typed AIR code: error 'error 'error 'error 'error 'error 'use of undeclared variable self@' in expression 'self@'' in expression '(%Poly%S. self@)'' in expression '(S./S/s (%Poly%S. self@))'' in expression '(Poly%pervasive.seq.Seq<nat.>. (S./S/s (%Poly%S. self@)))'' in expression '(pervasive.seq.Seq.index.? NAT (Poly%pervasive.seq.Seq<nat.>. (S./S/s (%Poly%S. self@)))
 idx@
)'' in expression '(%I (pervasive.seq.Seq.index.? NAT (Poly%pervasive.seq.Seq<nat.>. (S./S/s (%Poly%S. self@)))
  idx@
))'', rust_verify/src/verifier.rs:371:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
verification results:: verified: 0 errors: 0
```

cc @parno

**Comment by parno** [comment]
@matthias-brun Thanks for the succinct bug report!

**Comment by Chris-Hawblitzel** [comment]
I ran across a variant of the regression test that was added for https://github.com/verus-lang/verus/pull/297 that still fails.  Replacing `ext_equal` with `==` in this test causes a `Found an interpreter expression ... outside the interpreter` panic:

```
        // GitHub issue 294: Make sure we don't let local variable names
        // leak out, even when a sequence function can't concretely evaluate
        spec fn f(s: &Seq<int>, idx: int) -> int {
            s.index(idx)
        }

        spec fn g(a: &Seq<int>, b: Seq<int>) -> Seq<int> {
            a.add(b)
        }

        // Try various combinations of simplified sequence
        // literal and symbolic values
        proof fn test_github_issue_294() {
            let x = seq![1, 2, 3];
            assert(f(&x, 0) == 1) by (compute);
            let y = seq![4, 5, 6];
            assert(g(&x, y).len() == 6) by (compute);
            assert({
                let z = seq![7, 8, 9];
                g(&x, z).len() == 6 &&
                g(&z, x).len() == 6
            }) by (compute);
            assert({
                let z = seq![4int, 5int, 6int];
                y == z &&
                z == y
            }) by (compute);
        }
```

**Comment by parno** [comment]
Closed by #568
